Assets/Scripts/DraggableBlock.cs

```diff
@@ -12,9 +12,18 @@ public class DraggableBlock : MonoBehaviour
     Block block;
     bool isDraggable = false;
 
+    Tile[,] tiles;
+    GameObject prefabTile;
+    Grid grid;
+    DraggableObject draggableObject;
+
+    //Copies the data of a Block into this DraggableBlock’s contained Block
     public void SetBlock(Block copiedBlock)
     {
-
+        GameObject tileObject = Instantiate(prefabTile);
+        tiles = TileUtil.CreateTileArray(prefabTile, tileObject.transform.position, copiedBlock.GetWidth(), copiedBlock.GetHeight());
+        block = new Block(copiedBlock.GetWidth(), copiedBlock.GetHeight());
```

Comment:
After creating this block, set its TileTypes to match that of the copiedBlock by using block.Fill(copiedBlock.GetTileType(row, col)). Afterwards, call UpdateTiles() immediately before the UpdateAvailableSpaces() call.